src/main/java/io/appium/java_client/flutter/FlutterDriver.java

```diff
@@ -0,0 +1,23 @@
+package io.appium.java_client.flutter;
+
+import org.openqa.selenium.WebDriver;
+
+/**
+ * The {@code FlutterDriver} interface represents a driver that controls interactions with
+ * Flutter applications, extending WebDriver and providing additional capabilities for
+ * interacting with Flutter-specific elements and behaviors.
+ *
+ * <p> This interface serves as a common entity for drivers that support Flutter applications
+ * on different platforms, such as Android and iOS. </p>
+ *
+ * @see WebDriver
+ * @see SupportsGestureOnFlutterElements
+ * @see SupportsScrollingOfFlutterElements
+ * @see SupportsWaitingForFlutterElements
+ */
+public interface FlutterDriver extends
+        WebDriver,
```

Comment:
should we include WebDriver here?

Reply:
I was thinking on the line we can have a generic interface where users can use both flutter methods and underlying native methods without have to deal with multiple type castings.  Tests will look something like below

```java
FlutterDriver driver = new FlutterAndroidDriver();

driver.findElement(AppiumBy.id("some_id")).click();
driver.findElement(AppiumBy.flutterText("Login")).click();

driver.waitTillVisible(new WaitParameter().setLocator(AppiumBy.flutterText("Welcome")));
```

Without the webdriver interface

```java
Without wendriver interface added:

AndroidDriver driver = new FlutterAndroidDriver();
driver.findElement(AppiumBy.id("some_id")).click();
driver.findElement(AppiumBy.flutterText("Login")).click();

((FlutterDriver)driver).waitTillVisible(new WaitParameter().setLocator(AppiumBy.flutterText("Welcome")));
```

Reply:
thanks for the explanation. It makes sense to me now